src/PrestaShopBundle/Controller/Admin/Improve/International/CurrencyController.php

```diff
@@ -0,0 +1,293 @@
+        $gridPresenter = $this->get('prestashop.core.grid.presenter.grid_presenter');
+
+        return [
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->generateUrl('admin_currency_create'),
+                    'desc' => $this->trans('Add new currency', 'Admin.International.Feature'),
+                    'icon' => 'add_circle_outline',
+                ],
+            ],
+            'currencyGrid' => $gridPresenter->present($currencyGrid),
+        ];
+    }
+
+    /**
+     * Provides filters functionality.
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $definitionFactory = $this->get('prestashop.core.grid.definition.factory.currency');
+        $definitionFactory = $definitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+        $searchParametersForm = $gridFilterFormFactory->create($definitionFactory);
+        $searchParametersForm->handleRequest($request);
+
+        $filters = [];
+        if ($searchParametersForm->isSubmitted()) {
+            $filters = $searchParametersForm->getData();
+        }
+
+        return $this->redirectToRoute('admin_currency_index', ['filters' => $filters]);
+    }
+
+    /**
+     * Displays currency form.
+     *
+     * @AdminSecurity("is_granted('create', request.get('_legacy_controller'))", message="You do not have permission to add this.")
+     *
+     * @return RedirectResponse
+     */
+    public function createAction()
+    {
+        //todo: drop legacy and after having post add @DemoRestricted
+        $legacyLink = $this->getAdminLink('AdminCurrencies', [
+            'addcurrency' => 1,
+        ]);
+
+        return $this->redirect($legacyLink);
+    }
+
+    /**
+     * Displays currency form.
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", message="You do not have permission to edit this.")
+     *
+     * @param int $currencyId
+     *
+     * @return RedirectResponse
+     */
+    public function editAction($currencyId)
+    {
+        //todo: drop legacy and after having post add @DemoRestricted
+        $legacyLink = $this->getAdminLink('AdminCurrencies', [
+            'id_currency' => $currencyId,
+            'updatecurrency' => 1,
+        ]);
+
+        return $this->redirect($legacyLink);
+    }
+
+    /**
+     * Deletes currency.
+     *
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", message="You do not have permission to delete this.")
+     * @DemoRestricted(redirectRoute="admin_currency_index")
+     *
+     * @param int $currencyId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction($currencyId)
+    {
+        $commandBus = $this->get('prestashop.core.command_bus');
```

Comment:
you can do `$this->getCommandBus()->handle(new Command);` no need to save bus into variable.